pkg/specgen/generate/oci.go

```diff
@@ -105,7 +105,12 @@ func makeCommand(ctx context.Context, s *specgen.SpecGenerator, img *image.Image
 		entrypoint = newEntry
 	}
 
-	finalCommand = append(finalCommand, entrypoint...)
+	// Ignore empty fields in entrypoint
+	for _, field := range entrypoint {
+		if field != "" {
+			finalCommand = append(finalCommand, field)
+		}
+	}
```

Comment:
I think that's too much since we're skipping any fields which has some potential for side effects. Docker does the following:
```golang
// Reset the Entrypoint if it is [""]
if len(config.Entrypoint) == 1 && config.Entrypoint[0] == "" {
        config.Entrypoint = nil
}
if len(config.Entrypoint) == 0 && len(config.Cmd) == 0 {
        return fmt.Errorf("No command specified")
}
```

Reply:
An image with `Entrypoint: ["", "", "/bin/sh"]` IMHO is wrong and should be rejected. While `[""]` could be argued for being unset.